### DISCUSSION TITLE
Method to know all validated available addresses (or tags) in Siemens PLC

### DISCUSSION THREAD

**Discussion by Rahul14189** [gijzelaerr/python-snap7#464] [Q&A] [answered]
Hi, I am trying to access all available or configured address in Siemens PLC, I can see we have db_get(db_number: int),  but here we have to pass existing db_number,  however it does not fetch addresses for input, output, timer, merker and counter.

So as there is a method in allen bradley, get_tag_list() to get all available tag or address, do we have something similar in siemens python snap7 too !!

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. Unlike Allen Bradley's tag-based architecture, Siemens PLCs do not expose a tag list via the S7 protocol. You can use `list_blocks()` and `list_blocks_of_type()` to enumerate blocks.
